controllers/namespace_controller.go

```diff
@@ -165,11 +168,15 @@ func (r *NamespaceReconciler) ensureLabel(ns *corev1.Namespace, tenantName strin
 	return nil
 }
 
-func (r *NamespaceReconciler) updateTenantStatus(ns *corev1.Namespace, tenant *v1alpha1.Tenant) {
-	switch ns.Status.Phase {
-	case corev1.NamespaceTerminating:
-		r.removeNamespace(ns.Name, tenant)
-	case corev1.NamespaceActive:
-		r.addNamespace(ns.Name, tenant)
-	}
+func (r *NamespaceReconciler) updateTenantStatus(ns *corev1.Namespace, tenant *v1alpha1.Tenant) error {
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		switch ns.Status.Phase {
+		case corev1.NamespaceTerminating:
+			r.removeNamespace(ns.Name, tenant)
+		case corev1.NamespaceActive:
+			r.addNamespace(ns.Name, tenant)
+		}
+
+		return r.Client.Status().Update(context.TODO(), tenant, &client.UpdateOptions{})
```

Comment:
This is going to fix bug introduced by a3c667b2b9c97c4b2dfe6156f9016ab6b75a2a41 since we weren't updating anymore the Tenant status, causing a missing reconciliation.

Reply:
Resolved on HEAD with #43